docs/concepts/models.md

```diff
@@ -464,6 +464,13 @@ except ValidationError as e:
 If you want to validate serialized data in a format other than JSON, you should load the data into a dict yourself and
 then pass it to `model_validate`.
 
+!!! note
+    In v2.5.0 and above, Pydantic uses [`jiter`](https://docs.rs/jiter/latest/jiter/),
+    a fast and iterable JSON parser, to parse JSON data. Using `jiter` compared to `serde` results in modest performance
+    improvements that will get even better in the future. The `jiter` JSON parser is almost entirely compatible with
+    the `serde` JSON parser, with one noticeable enhancement being that `jiter` supports serialization of `inf` and `NaN` values.
+    In the future, `jiter` will enable support for localizable JSON error messages from JSON parsing errors.
+
```

Comment:
Happy to get some feedback on the phrasing of this - just a first pass, here.

Reply:
That's a good point. Given that we're moving this to its own concepts page, I think it makes sense to mention. If anyone is curious enough to find themselves on that page, it probably doesn't hurt to have the extra info.